File: AFTAS-backend/src/main/java/com/aftas/aftasbackend/model/entities/Hunting.java
package com.aftas.aftasbackend.model.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Builder
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "huntings")
public class Hunting {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer numberOfFish;

    @ManyToOne
    @JoinColumn(name = "fish_id")
    private Fish fish;

    @ManyToOne
    @JoinColumn(name = "competition_id")
    private Competition competition;

    @ManyToOne
    @JoinColumn(name = "member_id")
    private Member member;

}

File: AFTAS-frontend/aftas-frontend/src/app/models/hunting.ts
import { Competition } from "./competition";
import { Fish } from "./fish";
import { Member } from "./member";

export interface Hunting {
    id?: number;
    numberOfFish: number;
    fishId: number;
    competitionId: number;
    weight: number;
    memberId: number;
    fish?: Fish;
    competition?: Competition;
    member?: Member;
}
